### PR TITLE
Fix IMDSv2 URL when fetching credentials

### PR DESCRIPTION
**Pull request by setu4993**
I missed 1 change in the final version of https://github.com/minio/minio-py/pull/1437 where the URL to fetch the endpoint must include `"/latest/meta-data/iam/security-credentials/"` before the role name. This adds that back.

Closes https://github.com/minio/minio-py/issues/1459.

### CODE REVIEW

File: minio/credentials/providers.py [hunk 1]
```diff
@@ -504,22 +504,18 @@ def retrieve(self) -> Credentials:
...
             res = _urlopen(
                 self._http_client,
                 "GET",
-                urlunsplit(
-                    url_replace(
-                        urlsplit(url),
-                        path="/latest/meta-data/iam/security-credentials/",
-                    ),
-                ),
-                headers=headers,
```
Comment (setu4993) [inline minio/credentials/providers.py:517]: This was unnecessary so I simplified it with L507.

File: minio/credentials/providers.py [hunk 2]
```diff
@@ -504,22 +504,18 @@ def retrieve(self) -> Credentials:
...
+                url,
+                headers=headers
             )
             role_names = res.data.decode("utf-8").split("\n")
             if not role_names:
                 raise ValueError(f"no IAM roles attached to EC2 service {url}")
-            url += "/" + role_names[0].strip("\r")
+            url += role_names[0].strip("\r")
```
Comment (setu4993) [inline minio/credentials/providers.py:518]: The role gets added at the end of the URL that was updated on L507.

**Comment by setu4993** [comment]
@balamurugana : Fixes IMDSv2 that didn't work correctly. Only found out today.

File: minio/credentials/providers.py (hunk 2 above)
Comment (setu4993) [reply minio/credentials/providers.py:518]: This is also in the Go implementation here: https://github.com/minio/minio-go/blob/e337e77c06bbf8a65247fb018e59c1263e4d86e1/pkg/credentials/iam_aws.go#L400-L402

We just missed it in the revisions in https://github.com/minio/minio-py/pull/1437/.

**Comment by balamurugana** [comment]
I am closing on behalf of https://github.com/minio/minio-py/pull/1465

**Pull request closed without merging**